src/js/models/layerstructure.js

```diff
@@ -801,6 +801,19 @@ define(function (require, exports, module) {
     }));
 
     /**
+     * Given a set of layers, return only those layers which will support being exported
+     * This excludes background layers and empty layers
+     *
+     * @param {Immutable.Iterable.<Layer>} layers
+     * @return {Immutable.Iterable.<Layer>}
+     */
+    Object.defineProperty(LayerStructure.prototype, "filterExportable", objUtil.cachedLookupSpec(function (layers) {
+        return layers.filterNot(function (layer) {
+            return !layer.bounds || layer.bounds.empty || layer.isBackground || this.isEmptyGroup(layer);
```

Comment:
Are adjustment layers supported? I'd actually explicitly list all the layer kinds that aren't supported here. We're going to have to do something better than we're currently doing overall for unsupported documents...